### ISSUE TITLE
rule `restrict-dependency-ranges` don't output any error

### ISSUE THREAD

**Issue by gadzhimari** [JoshuaKGoldberg/eslint-plugin-package-json#1891]
Hi. I don't see any error that should be triggered by `restrict-dependency-ranges` rule. In my project I have the following .eslintrc file. The version of ESlint is 

```
"eslint": "^8.48.0",
"eslint-plugin-package-json": "^1.2.0",
```

Here my full config:

```
{
  "parser": "@typescript-eslint/parser", // Specifies the ESLint parser
  "extends": [
    "plugin:import/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "plugin:react/jsx-runtime",
    "plugin:@typescript-eslint/recommended",
    "prettier",
    "plugin:prettier/recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "env": {
    "browser": true,
    "es2021": true
  },
  "parserOptions": {
    "ecmaVersion": "latest", // Allows for the parsing of modern ECMAScript features
    "sourceType": "module", // Allows for the use of imports
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "plugins": ["react", "prettier", "package-json"],
  "rules": {
    // Place to specify ESLint rules. Can be used to overwrite rules
    // specified from the extended configs
    // e.g. "@typescript-eslint/explicit-function-return-type": "off",
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "import/prefer-default-export": "off",
    "import/no-unresolved": "off",
    "react/jsx-filename-extension": [
      1,
      {
        "extensions": [".js", ".jsx", ".tsx", ".ts"]
      }
    ],
    "react/function-component-definition": [
      2,
      {
        "namedComponents": "arrow-function"
      }
    ],
    "semi": ["error", "always"],
    "indent": "off",
    "linebreak-style": ["error", "unix"],
    "quotes": ["error", "single"],
    "no-multi-spaces": "error",
    "@typescript-eslint/no-unused-vars": "warn",
    "space-in-parens": "error",
    "no-multiple-empty-lines": "error",
    "@typescript-eslint/no-explicit-any": "warn",
    "prettier/prettier": "error",
    "@typescript-eslint/consistent-type-imports": [
      "warn",
      {
        "prefer": "type-imports"
      }
    ],
    "import/order": [
      "warn",
      {
        "groups": [["builtin", "external", "internal"], "type"],
        "newlines-between": "always",
        "alphabetize": {
          "order": "asc",
          "caseInsensitive": true
        }
      }
    ],
    "package-json/restrict-dependency-ranges": [
      "error",
      {
        "forDependencyTypes": ["devDependencies"],
        "rangeType": "pin"
      }
    ]
  },
  "settings": {
    "react": {
      "version": "detect"
    },
    "import/resolver": {
      "node": {
        "extensions": [".js", ".jsx", ".ts", ".tsx"],
        "moduleDirectory": ["node_modules", "src"]
      },
      "typescript": {}
    }
  }
}
```

<img width="2332" height="884" alt="Image" src="https://github.com/user-attachments/assets/27c87d8d-6c4a-4da5-a86a-3932813eef05" />

**Comment by michaelfaith** [comment]
Hi there. We do have an issue template that we ask you use. Please refile using the issue template. This doesn't have all the information we need to help you.

**Comment by gadzhimari** [comment]
[Here](https://github.com/michaelfaith/eslint-plugin-package-json/issues/1893) I create new one
